src/app/components/turnos/dar-turnos/dar-sobreturno.component.ts

```diff
@@ -0,0 +1,326 @@
+            this.fin = new Date(this.hoy.setHours(this.agenda.horaFin.getHours(), this.agenda.horaFin.getMinutes(), 0, 0));
+            if (this.agenda.tipoPrestaciones.length === 1) {
+                this.tipoPrestacion = this.agenda.tipoPrestaciones[0];
+            }
+            if (agenda.link) {
+                this.turno_link = agenda.link;
+            }
+        });
+
+        this.servicePaciente.getById(this.idPaciente).subscribe(res => {
+            this.paciente = res;
+            this.loadObraSocial(this.paciente);
+        });
+
+        this.carpetaEfector = {
+            organizacion: {
+                _id: this.auth.organizacion.id,
+                nombre: this.auth.organizacion.nombre
+            },
+            nroCarpeta: ''
+        };
+    }
+
+
+    afterCreateUpdate(paciente) {
+        this.showCreateUpdate = false;
+        if (paciente) {
+            this.servicePaciente.getById(paciente.id).subscribe(
+                pacienteMPI => {
+                    this.paciente = pacienteMPI;
+                    this.verificarTelefono(pacienteMPI);
+                });
+        }
+    }
+
+    loadObraSocial(paciente) {
+        // TODO: si es en colegio médico hay que buscar en el paciente
+        if (!paciente || !paciente.documento) {
+            return;
+        }
+        this.obraSocialService.getObrasSociales(paciente.documento).subscribe(resultado => {
+            if (resultado.length) {
+                this.obraSocialPaciente = resultado.map((os: any) => {
+                    let osPaciente;
+
+                    if (os.nombre) {
+                        osPaciente = {
+                            'id': os.nombre,
+                            'label': os.nombre
+                        };
+                    } else {
+                        osPaciente = {
+                            'id': os.financiador,
+                            'label': os.financiador
+                        };
+                    }
+                    return osPaciente;
+                });
+                this.modelo.obraSocial = this.obraSocialPaciente[0].label;
+            }
+            this.obraSocialPaciente.push({ 'id': 'prepaga', 'label': 'Prepaga' });
+        });
+    }
+
+    seleccionarObraSocial(event) {
+        if (event.value === 'prepaga') {
+            this.obraSocialService.getPrepagas().subscribe(prepagas => {
+                this.showListaPrepagas = true;
+                this.prepagas = prepagas;
+            });
+        } else {
+            this.showListaPrepagas = false;
+        }
+        this.modelo.obraSocial = event && event.value;
+    }
+
+    // Operaciones con carpetaPaciente
+
+    // Se busca el número de carpeta de la Historia Clínica en papel del paciente
+    // a partir del documento y del efector
+    obtenerCarpetaPaciente() {
+        let indiceCarpeta = -1;
+        if (this.paciente.carpetaEfectores.length > 0) {
```

Comment:
No es necesario tener que aclarar en un .length que tiene que ser mayer a cero. 
Con hacer if (this.paciente.carpetaEfectores.length)  ya se esta evaluando si es mayor a cero.

Reply:
Cambios realizados!